### PR TITLE
[GUVNOR-2313]People perspective - Follow button doesn't change to unfollow after user follows someone

### PR DESCRIPTION
**Pull request by ederign**
Issue
=====
When user searches for some other user in People perspective and clicks the 'Follow' button, the button doesn't change to 'Unfollow' as it did in previous version.

Fix
===

Refresh the user cache each time that the user are followed/unfollowed.

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/test/org/kie/workbench/common/screens/social/hp/client/userpage/UserHomePageSidePresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        presenter.users = new HashMap<String, SocialUser>(  );
+
+        when( socialUserRepositoryAPI.findSocialUser( bento.getUserName() ) )
+                .thenReturn( bento );
+
+        presenter.unfollowUser( bento );
+
+        verify( selectEvent).fire( any( UserHomepageSelectedEvent.class ) );
+        assertEquals( bento, presenter.users.get( "bento" ) );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/test/org/kie/workbench/common/screens/social/hp/client/userpage/UserHomePageSidePresenterTest.java]: Why is the assertion in "unfollow" the same as that in "follow"?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/test/org/kie/workbench/common/screens/social/hp/client/userpage/UserHomePageSidePresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Before
+    public void setup() {
+        presenter = new UserHomePageSidePresenter();
+
+        socialUserServiceAPICaller = new CallerMock<SocialUserServiceAPI>( socialUserServiceAPI );
+
+        socialUserRepositoryAPICaller = new CallerMock<SocialUserRepositoryAPI>( socialUserRepositoryAPI );
+
+        presenter.socialUserService = socialUserServiceAPICaller;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/test/org/kie/workbench/common/screens/social/hp/client/userpage/UserHomePageSidePresenterTest.java]: Constructor injection might have been nicer; but it's only a style/preference issue.. technically not wrong - just more error prone.

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/test/org/kie/workbench/common/screens/social/hp/client/userpage/UserHomePageSidePresenterTest.java (hunk 1 above)
Comment (ederign) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/test/org/kie/workbench/common/screens/social/hp/client/userpage/UserHomePageSidePresenterTest.java]: I'm asserting that after the follow/unfollow backend call we refresh view cache.

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
Thanks.

master: 90643bd7ffc78e229751536c22dddc54b9f015ac

**Pull request closed without merging**